encord/dataset.py

```diff
@@ -0,0 +1,176 @@
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional, TextIO, Union
+
+from encord.client import EncordClientDataset
+from encord.orm.dataset import AddPrivateDataResponse, DataRow
+from encord.orm.dataset import Dataset as OrmDataset
+from encord.orm.dataset import ImageGroupOCR, StorageLocation
+
+
+class Dataset:
+    """
+    Access dataset related data and manipulate the dataset.
+    """
+
+    def __init__(self, client: EncordClientDataset):
+        self._client = client
+        self._dataset_instance = None
+
+    @property
+    def title(self) -> str:
+        dataset_instance = self._get_dataset_instance()
+        return dataset_instance.title
+
+    @property
+    def description(self) -> str:
+        dataset_instance = self._get_dataset_instance()
+        return dataset_instance.description
+
+    @property
+    def storage_location(self) -> StorageLocation:
+        dataset_instance = self._get_dataset_instance()
+        return dataset_instance.storage_location
+
+    @property
+    def data_rows(self) -> List[DataRow]:
+        dataset_instance = self._get_dataset_instance()
+        return dataset_instance.data_rows
+
+    def refetch_data(self) -> None:
+        """
+        The Dataset class will only fetch its properties once. Use this function if you suspect the state of those
+        properties to be dirty.
+        """
+        self._dataset_instance = self.get_dataset()
+
+    def get_dataset(self) -> OrmDataset:
+        """
+        This function is exposed for convenience. You are encouraged to use the property accessors instead.
+        """
+        return self._client.get_dataset()
+
+    def upload_video(self, file_path: str):
+        """
+        Upload video to Encord storage.
+
+        Args:
+            self: Encord client object.
+            file_path: path to video e.g. '/home/user/data/video.mp4'
+
+        Returns:
+            Bool.
+
+        Raises:
+            UploadOperationNotSupportedError: If trying to upload to external
+                                              datasets (e.g. S3/GPC/Azure)
+        """
+        return self._client.upload_video(**{k: v for k, v in locals().items() if k != "self"})
```

Comment:
This looks interesting, also quite dangerous. I think passing the parameters explicitly the way to go, or doing something with *args, **kwargs and typehints

Reply:
sounds good, I'll pass them explicitly to not mess with the type signature for sphinx